webserver/app/helpers/kubernetes.py

```diff
@@ -64,27 +64,34 @@ def create_pod_spec(self, pod_spec:dict):
         # Create a dedicated VPC for each task so that we can keep results indefinitely
         self.create_persistent_storage(pod_spec["name"], pod_spec["labels"])
         pvc_name = f"{pod_spec["name"]}-volclaim"
+        pvc = client.V1PersistentVolumeClaimVolumeSource(claim_name=pvc_name)
 
-        vol_mount = client.V1VolumeMount(
-            mount_path=pod_spec["mount_path"],
-            sub_path=pod_spec['labels']['task_id'],
-            name="data"
-        )
+        vol_mounts = []
+
+        for mount_name, mount_path in pod_spec.get("mount_path", {}).items():
```

Comment:
could you define `pod_spec['labels']['task_id']}` as a named variable before using it in the string?

It took me a little while to understand what this loop was supposed to be doing because of it :S 

Silly, but...

Reply:
Good to merge though.

Reply:
Sure, will set it to a separate var and sprinkle some comments around